internal/pkg/cli/job_logs.go

```diff
@@ -151,38 +157,43 @@ func (o *jobLogsOpts) Ask() error {
 	return o.validateAndAskJobEnvName()
 }
 
-func (o *jobLogsOpts) validateOrAskApp() error {
-	if o.appName != "" {
-		_, err := o.configStore.GetApplication(o.appName)
+// Execute outputs logs of the job.
+func (o *jobLogsOpts) Execute() error {
+	if err := o.initLogsSvc(); err != nil {
 		return err
 	}
-	app, err := o.sel.Application(jobAppNamePrompt, svcAppNameHelpPrompt)
-	if err != nil {
-		return fmt.Errorf("select application: %w", err)
+	eventsWriter := logging.WriteHumanLogs
+	if o.shouldOutputJSON {
+		eventsWriter = logging.WriteJSONLogs
+	}
+	var limit *int64
+	if o.limit != 0 {
+		limit = aws.Int64(int64(o.limit))
 	}
-	o.appName = app
-	return nil
-}
 
-func (o *jobLogsOpts) validateAndAskJobEnvName() error {
-	if o.envName != "" {
-		if _, err := o.getTargetEnv(); err != nil {
-			return err
-		}
+	// By default, only display the logs of the last execution of the job.
+	logStreamLimit := defaultJobLogExecutionLimit
+	if o.last != 0 {
+		logStreamLimit = o.last
 	}
 
-	if o.name != "" {
-		if _, err := o.configStore.GetJob(o.appName, o.name); err != nil {
-			return err
-		}
+	if o.includeStateMachineLogs {
+		logStreamLimit *= 2
```

Comment:
Is this a safe assumption that we can make 💭 If so we might need to add some comments explaining the reason. Also I feel like we should deal with the logic within the logging pkg, because otherwise we are passing duplicated info around.

Reply:
Good idea. I guess we can use more jobby abstractions in `logging` since it's not the low-level aws package.